src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -176,61 +207,214 @@ def __blob_is_not_complete(metadata):
+        for name_value in secret_build_args:
             name, value = name_value.split('=', 1)
             build_arguments.append(BuildArgument(name, value, True))
 
-    build_request = QueueBuildRequest(
-        image_name=image_name,
-        source_location=source_location,
-        build_parameters=build_parameters,
-        is_push_enabled=is_push_enabled,
-        timeout=timeout,
-        platform=platform,
-        build_arguments=build_arguments)
-
-    result = LongRunningOperation(cmd.cli_ctx)(client.queue(
-        build_request=build_request, resource_group_name=resource_group_name, registry_name=registry_name))
-
-    if isinstance(result, Build):
-        print("Successfully queued a build with build-id: {}. Starting to stream the logs...".format(result.build_id))
-        return acr_build_show_logs(cmd, client, registry_name, result.build_id, resource_group_name)
+    try:
+        build_request = QueueBuildRequest(
+            image_name=image_name,
+            source_location=source_location,
+            build_parameters=build_parameters,
+            is_push_enabled=is_push_enabled,
+            timeout=timeout,
+            platform=platform,
+            build_arguments=build_arguments)
+
+        result = LongRunningOperation(cmd.cli_ctx)(client.queue(
+            build_request=build_request, resource_group_name=resource_group_name, registry_name=registry_name))
+
+        print("Successfully queued a build with build-id: {}.".format(result.build_id))
+
+        if show_logs == 'true':
+            print("Starting to stream the logs...")
+            return acr_build_show_logs(cmd, client, registry_name, result.build_id, resource_group_name)
+    except Exception as err:
+        raise CLIError(err)
+
+
+def _check_remote_source_code(source_location):
+
+    # NOTE: acr_build doesn't support git repo url
+    if source_location.lower().startswith("git://"):
+        return source_location
+
+
+    response = requests.head(source_location)
+    if response.status_code < 400:
+        return source_location
+    else:
+        raise CLIError("'{}' doesn't exist.".format(source_location))
+
+
+def _check_image_name(image_name):
+    
+    # referenc: https://github.com/docker/distribution/tree/master/reference
+
+    if image_name == "":
+        raise CLIError("'--tag' value should not be empty.")
+    
+    tokens = image_name.split(':')
+    if(len(tokens) > 2):
+        raise CLIError("'--tag' value should be repository-name and optionally a tag in the 'repository-name:tag' format")
+
+    # check repository-name    
+    repository_name = tokens[0]
+    if len(repository_name) > 255:
+        raise CLIError("The repository-name of '--tag' value should be no more than 255 characters.")
     else:
-        # Maybe just raise error?
-        return result
+        if re.match(r"^[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?$", repository_name) is None:
```

Comment:
IMO an overkill. Client app should avoid validation. Let the server side validate.

Reply:
I'm ok with this since we can avoid a round trip and the image name regex is more of a spec. 

---
In reply to: [168359952](https://github.com/AppPlatPerf/azure-cli/pull/60#discussion_r168359952) [](ancestors = 168359952)

Reply:
Also in reply to https://github.com/AppPlatPerf/azure-cli/pull/60/files#r168366695

I had a strong opinion on this because of the issues we had in the past (e.g., we even had inconsistency issues with resource name length). It's a lot of extra work and user disappointment as well as effort on us to ask users to use a specific version of CLI (or clear their browser cache to load a newer version of portal).

I would suggest two other approaches here:

1. If it's a spec, move it to swagger to have a single source of truth that all SDKs/clients/RP (and third party apps if they care about consistency/correctness) can follow. CLI custom code should never be bothered with such a validation (especially considering the regex here is not a simple one).

2. Forget about validations in SDKs/clients and just use RP as a single source of truth. I would prefer this one since updating RP is much easier than updating swagger.